lib/SILGen/SILGenExpr.cpp

```diff
@@ -2293,6 +2293,20 @@ RValue RValueEmitter::visitRebindSelfInConstructorExpr(
   return SGF.emitEmptyTupleRValue(E, C);
 }
 
+static bool isNullableTypeInC(Type ty) {
+  if (ty->hasReferenceSemantics())
+    return true;
+  
+  auto &C = ty->getASTContext();
+  auto nom = ty->getAnyNominal();
+  if (!nom)
+    return false;
+
+  return nom == C.getUnsafePointerDecl()
+    || nom == C.getUnsafeMutablePointerDecl()
+    || nom == C.getAutoreleasingUnsafeMutablePointerDecl();
+}
```

Comment:
Unfortunately it's not defined in terms of Swift types. https://github.com/apple/swift/blob/master/lib/ClangImporter/ClangImporter.cpp#L1533

…ugh, it looks like that code is wrong too.